source/agora/node/Node.d

```diff
@@ -86,4 +93,30 @@ public class Node (Network) : API
 
         return config;
     }
+
+    /***************************************************************************
+
+        Receive a message.
+
+        API:
+            PUT /message
+
+        Returns:
+            Return true if this message was first received at this node.
+
+        Params:
+          msg = the message received
+
+    ***************************************************************************/
+    
+    public override bool setMessage(Hash msg) @safe
+    {
+		return this.gossip.receiveMessage(msg);
+    }
+
+    /// GET: /hasMessage
+    public override bool getHasMessage(Hash msg) @safe
```

Comment:
@Geod24 does vibe.d have some form of user attributes so we can do something like `@get bool hasMessage`? Because "getHasMessage" sounds really weird in English.

Reply:
@Geod24 ping

Reply:
It does and it's in the documentation: http://vibed.org/api/vibe.web.rest/
It's `@method(...)`

Reply:
source/agora/common/API.d(117,13): Deprecation: vibe.web.rest.HTTPMethod is not visible from module API

Reply:
`    @method(HTTPMethod.GET)
    public bool hasMessage(Hash msg);`

Reply:
For the warning above you need to import `vibe.http.common`